lib/factory_girl/linter.rb

```diff
@@ -1,11 +1,12 @@
 module FactoryGirl
   class Linter
-    def self.lint!(factories_to_lint)
-      new(factories_to_lint).lint!
+    def self.lint!(factories_to_lint, options)
```

Comment:
@ndp from an interface perspective, yes; you're passing a flag to a class method to do something additional conditionally, and the method to calculate factories has a second round of behavior that's now significantly more complex to test.

What if a "factory" in the sense here is an array of size N, where N can be 1 (just the factory) or 2 (factory plus a trait) or 4 (if we want to get into trait combinations and the factory has three traits) - and treat each as its own factory that can be built (with `FactoryGirl.create(*factory_and_traits)`)?

I still think there's value in opting into trait linting - since it's going to add a lot of additional time, depending on the object model - but I'm not sure where the best place to handle that would be. Possibly configuration?